src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -194,68 +153,59 @@ protected override void BeginProcessing()
         protected override void ProcessRecord()
         {
             bool result = true;
-            string jsonToParse = string.Empty;
-
-            if (Json != null)
-            {
-                jsonToParse = Json;
-            }
-            else if (Path != null)
-            {
-                string resolvedPath = PathUtils.ResolveFilePath(Path, this, _isLiteralPath);
-
-                if (!File.Exists(resolvedPath))
-                {
-                    ItemNotFoundException exception = new(
-                        Path,
-                        "PathNotFound",
-                        SessionStateStrings.PathNotFound);
-
-                    ThrowTerminatingError(exception.ErrorRecord);
-                }
-
-                jsonToParse = File.ReadAllText(resolvedPath);
-            }
-
-            if (string.IsNullOrWhiteSpace(jsonToParse))
-            {
-                WriteObject(false);
-                return;
-            }
 
             try
             {
-                var parsedJson = JToken.Parse(jsonToParse);
+                var parsedJson = JsonNode.Parse(Json);
 
                 if (_jschema != null)
                 {
-                    var errorMessages = _jschema.Validate(parsedJson);
-                    if (errorMessages != null && errorMessages.Count != 0)
-                    {
-                        result = false;
-
-                        Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
-
-                        foreach (var message in errorMessages)
+                        var validationResults = _jschema.Validate(parsedJson, new ValidationOptions { OutputFormat = OutputFormat.Basic });
+                        result = validationResults.IsValid;
+                        if (!result)
                         {
-                            ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
-                            errorRecord.ErrorDetails = new ErrorDetails(message.ToString());
-                            WriteError(errorRecord);
+                            Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
+
+                            if (validationResults.Message != null)
+                            {
+                                ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchemaDetailed", ErrorCategory.InvalidData, null);
+                                errorRecord.ErrorDetails = new ErrorDetails(typeof(TestJsonCommand).Assembly, "TestJsonCmdletStrings", "InvalidJsonAgainstSchemaDetailed", validationResults.Message, validationResults.InstanceLocation);
+                                WriteError(errorRecord);
+                            }
```

Comment:
What does the error look like when rendered in terminal? I wonder what the error message that will be displayed by default in PowerShell (concise error view by default).
Instead of creating the exception `Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema)`, why not just have the following in the `if` block?
```
if (validationResults.Message != null)
{
    Exception exception = new(string.Format(TestJsonCmdletStrings.InvalidJsonAgainstSchemaDetailed, validationResults.Message, validationResults.InstanceLocation));
    ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchemaDetailed", ErrorCategory.InvalidData, null);
    WriteError(errorRecord);
}
```

Reply:
The way I have it was the way I saw it in other areas of the code, probably.  I'm not tied to it.

Reply:
Could you please share a few screenshots for the errors created in this code path? I'd like to see the error message that get rendered by default to the terminal. I'm worried that the detailed information won't be shown by default.

When there are multiple error records written out, it's OK to have multiple exceptions for them.

Reply:
The idea is that it's better for users to see the detailed error messages directly, rather than requiring them to capture the error record and then access properties within it to see the detailed errors.

Reply:
Honestly, I don't know how to run it locally. I've been relying on unit tests.

Reply:
Schema:

```json
{
    "type": "object",
    "properties": {
        "foo": { "type": "string" },
        "bar": { "type": "integer" }
    },
    "required": ["foo", "bar"]
}
```

Instance:

```json
{
    "foo": true
}
```

JSON Schema output:

```json

{
  "valid": false,
  "keywordLocation": "#",
  "instanceLocation": "#",
  "errors": [
    {
      "valid": false,
      "keywordLocation": "#/properties/foo/type",
      "instanceLocation": "#/foo",
      "error": "Value is \"boolean\" but should be \"string\""
    },
    {
      "valid": false,
      "keywordLocation": "#/required",
      "instanceLocation": "#",
      "error": "Required properties [\"bar\"] were not present"
    }
  ]
}
```

Powershell output:

```
PS C:\projects\PowerShell\src\powershell-win-core\bin\Debug\net8.0\win7-x64\publish> Test-json '{ "foo": true }' -schemaFile schema.json
Test-Json: The JSON is not valid with the schema.
Test-Json: The JSON is not valid with the schema.
False
```

Looks like it doesn't output the errors.

Reply:
If I use what you have:

```
PS C:\projects\PowerShell\src\powershell-win-core\bin\Debug\net8.0\win7-x64\publish> Test-json '{ "foo": true }' -schemaFile schema.json
Test-Json: The JSON is not valid with the schema: Value is "boolean" but should be "string" at #/foo
Test-Json: The JSON is not valid with the schema: Required properties ["bar"] were not present at #
False
```

Reply:
I'll push that. 😄